### ISSUE TITLE
Support client-only directives

### ISSUE THREAD

**Issue by nandorojo** [0no-co/GraphQLSP#142]
Is there a recommended way to get support for something like `@populate`?

<img width="754" alt="image" src="https://github.com/0no-co/GraphQLSP/assets/13172299/4477937f-db5d-482b-a3a6-5cbcf1c028df">

**Comment by JoviDeCroock** [comment]
Changed the title, basically we would want a way to support `client-only` directives. Common ones can be found like `@optional`, `@required`, `@argumentDefinitions`, ...

Additionally I'm spotting an offset bug with the line indicator for `@populate` which is why I also added the bug label. Can extract that later but most likely has to do with [this line](https://github.com/0no-co/GraphQLSP/blob/main/packages/graphqlsp/src/diagnostics.ts#L155-L157) as we need to offset 5 characters it looks like to get to the proper squiggly under `@populate`.... Not sure where those 5 characters come from yet but to investigate.
